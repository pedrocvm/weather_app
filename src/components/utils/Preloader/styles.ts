import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: rgb(0 0 0 / 80%);
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    pointer-events: none;
    z-index: ${theme.layers.overlay};
    ${theme.flex('row', 'center', 'center')};
  `}
`;
