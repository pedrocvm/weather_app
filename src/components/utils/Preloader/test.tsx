import React from 'react';
import { renderWithTheme } from 'utils/tests/helpers';

import Preloader from '.';

describe('<Preloader />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Preloader />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
