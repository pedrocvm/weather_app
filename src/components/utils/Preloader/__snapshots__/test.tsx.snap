// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> should render correctly 1`] = `
.c0 {
  background: rgb(0 0 0 / 80%);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  z-index: 30;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0;
}

<div
  class="c0"
>
  <svg
    height="150px"
    preserveAspectRatio="xMidYMid"
    style="margin: auto; display: block;"
    viewBox="0 0 100 100"
    width="150px"
  >
    <circle
      cx="50"
      cy="50"
      fill="none"
      r="0"
      stroke="#f5542e"
      stroke-width="2"
    >
      <animate
        attributeName="r"
        begin="0s"
        calcMode="spline"
        dur="1s"
        keySplines="0 0.2 0.8 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="0;40"
      />
      <animate
        attributeName="opacity"
        begin="0s"
        calcMode="spline"
        dur="1s"
        keySplines="0.2 0 0.8 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </circle>
    <circle
      cx="50"
      cy="50"
      fill="none"
      r="0"
      stroke="#f2c327"
      stroke-width="2"
    >
      <animate
        attributeName="r"
        begin="-0.5s"
        calcMode="spline"
        dur="1s"
        keySplines="0 0.2 0.8 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="0;40"
      />
      <animate
        attributeName="opacity"
        begin="-0.5s"
        calcMode="spline"
        dur="1s"
        keySplines="0.2 0 0.8 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </circle>
  </svg>
</div>
`;
